test/integration/requestid_test.go

```diff
@@ -287,3 +292,30 @@ func newAuthorizingServer(t *testing.T, mca *minica.CA) *httptest.Server {
 
 	return srv
 }
+
+func requireCAServerToBeAvailable(t *testing.T, address string, timeout time.Duration) {
+	t.Helper()
+
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	for !canConnect(ctx, address) {
+		select {
+		case <-ctx.Done():
+			require.FailNow(t, fmt.Sprintf("CA server failed to start at https://%s within %s", address, timeout.String()))
+		case <-time.After(100 * time.Millisecond):
+		}
+	}
+}
+
+func canConnect(ctx context.Context, address string) bool {
+	d := net.Dialer{Timeout: 5 * time.Second}
+	conn, err := d.DialContext(ctx, "tcp", address)
+	if err != nil {
+		return false
+	}
+
+	conn.Close()
+
+	return true
```

Comment:
It's faster if you just re-dial every 100ms, since you avoid running into TCP's exponential backoff.

Also, if the server isn't bound to the port yet, the kernel might send you a RST in response to the SYN immediately, which would make `DialContext` fail immediately.

Reply:
Returning an error immediately is OK, and kind of what I want. The CA server is started (or not) in a different goroutine. The goal is to wait for it to become available before continuing the test, and a failure to connect here indicates it's not yet available.